src/documents/management/commands/document_correspondents.py

```diff
@@ -0,0 +1,45 @@
+from django.core.management.base import BaseCommand
+
+from documents.models import Document, Correspondent
+
+from ...mixins import Renderable
+
+
+class Command(Renderable, BaseCommand):
+    help = """
+        Using the current set of correspondent rules, apply said rules to all
+        documents in the database, effectively allowing you to back-tag all
+        previously indexed documents with correspondent created (or modified)
+        after their initial import.
+    """.replace("    ", "")
+
+    def __init__(self, *args, **kwargs):
+        self.verbosity = 0
+        BaseCommand.__init__(self, *args, **kwargs)
+
+    def handle(self, *args, **options):
+
+        self.verbosity = options["verbosity"]
+
+        for document in Document.objects.all():
+            # No matching correspondents, so no need to continue
+            if document.correspondent:
+                continue
+
+            potential_correspondents = list(
+                Correspondent.match_all(document.content))
+            if not potential_correspondents:
+                continue
+
+            potential_count = len(potential_correspondents)
+
+            selected = potential_correspondents[0]
```

Comment:
I'm not thrilled about this decision here.  Imagine a situation where a user had 100 files that they haven't yet given a correspondent to.  They then create new correspondent matching rules for both their bank and for their ISP and run this script only to realise that they have their bank name on all of their ISP's invoices and now everything has be arbitrarily labelled with the wrong correspondent and there's no easy way to fix it.

A suggestion:

Why not draw a simple text menu + select option for cases where multiple correspondents match?  I'm thinking of something like this:

```
The file http://localhost:8000/fetch/doc/282 matches multiple potential correspondents.  Which of the following would you like to assign to it?

0. None of these, leave it empty
1. Telus Mobility
2. Scotiabank

Your selection[0]: 
```

Additionally, I'd like to have some sort of standardised log output for what goes down should the user screw up so they can then use that log output as a way to batch-fix whatever they've done.  Something like this:

```
Document 282: "ISP Bill - utilities.pdf" was given the correspondent id 7: "Telus Mobility"
```